libraries/AP_HAL/tests/test_prescaler.cpp

```diff
@@ -0,0 +1,57 @@
+#include <AP_gtest.h>
+#include <AP_HAL/HAL.h>
+#include <AP_HAL/RCOutput.h>
+
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();
+
+class PrescalerParameterizedTestFixture :public ::testing::TestWithParam<uint32_t> {
+protected:
+    void test_prescaler(uint32_t clock, uint32_t target_rate, bool is_dshot)
+    {
+        const uint32_t prescaler = AP_HAL::RCOutput::calculate_bitrate_prescaler(clock, target_rate, true, is_dshot);
+        // we would like at most a 1% discrepancy in target versus actual
+        const float rate_delta = fabsf(float(clock / prescaler) - target_rate) / target_rate;
+        EXPECT_TRUE(rate_delta < 0.01f);
+    }
+};
+
+TEST_P(PrescalerParameterizedTestFixture, DShot150) {
+    test_prescaler(GetParam(), 150000, true);
+}
+
+TEST_P(PrescalerParameterizedTestFixture, DShot300) {
+    test_prescaler(GetParam(), 300000, true);
+}
+
+TEST_P(PrescalerParameterizedTestFixture, DShot600) {
+    test_prescaler(GetParam(), 600000, true);
+}
+
+TEST_P(PrescalerParameterizedTestFixture, DShot1200) {
+    test_prescaler(GetParam(), 1200000, true);
+}
+
+TEST_P(PrescalerParameterizedTestFixture, Passthrough) {
+    test_prescaler(GetParam(), 192000, false);
```

Comment:
one too many zeros

Reply:
It's actually correct - the bit width is 10 so the actual frequency is 19200 * 10